src/core/stream_recv.c

```diff
@@ -869,28 +870,55 @@ QuicStreamRecvFlush(
         return;
     }
 
+    //
+    // Have 3 buffers one the stack, it's enough for most use cases.
+    //
+    QUIC_BUFFER StackRecvBuffers[3];
+    QUIC_BUFFER *RecvBuffers = StackRecvBuffers;
+    uint32_t RecvBufferCount = ARRAYSIZE(StackRecvBuffers);
+
     BOOLEAN FlushRecv = TRUE;
     while (FlushRecv) {
         CXPLAT_DBG_ASSERT(!Stream->Flags.SentStopSending);
 
-        QUIC_BUFFER RecvBuffers[3];
         QUIC_STREAM_EVENT Event = {0};
         Event.Type = QUIC_STREAM_EVENT_RECEIVE;
-        Event.RECEIVE.BufferCount = ARRAYSIZE(RecvBuffers);
-        Event.RECEIVE.Buffers = RecvBuffers;
+        Event.RECEIVE.BufferCount = ARRAYSIZE(StackRecvBuffers);
+        Event.RECEIVE.Buffers = StackRecvBuffers;
 
         //
         // Try to read the next available buffers.
         //
         BOOLEAN DataAvailable = QuicRecvBufferHasUnreadData(&Stream->RecvBuffer);
         if (DataAvailable) {
+            uint32_t NumBuffersNeeded = QuicRecvBufferReadBufferNeededCount(&Stream->RecvBuffer);
+            if (NumBuffersNeeded > RecvBufferCount) {
```

Comment:
This is annoying to have to do.... I wonder if there's a way we can eliminate this. We could only indicate at most 3 at a time...

Reply:
After giving it some more thought, I think I would rather keep it as it is unless we have a reason to think it has a negative impact.

Reply:
Why? I'd rather reduce the code complexity as much as possible. The assert path is preferrable IMO.

Reply:
Ignore the other comment. We can keep this as is.